src/components/character-count/character-count.js

```diff
@@ -0,0 +1,192 @@
+    countElement.setAttribute('aria-describedby', this.describedByInfo)
+    countMessage = document.getElementById(elementId + '-info')
+  }
+  return countMessage
+}
+
+// Bind input propertychange to the elements and update based on the change
+CharacterCount.prototype.bindChangeEvents = function () {
+  var $textarea = this.$textarea
+  $textarea.addEventListener('keyup', this.updateCountMessage.bind(this))
+
+  // Bind focus/blur events to start/stop polling
+  $textarea.addEventListener('focus', this.handleFocus.bind(this))
+  $textarea.addEventListener('blur', this.handleBlur.bind(this))
+}
+
+// Speech recognition software such as Dragon NaturallySpeaking will modify the
+// fields by directly changing its `value`. These changes don't trigger events
+// in JavaScript, so we need to poll to handle when and if they occur.
+CharacterCount.prototype.checkIfValueChanged = function () {
+  if (!this.oldValue) this.oldValue = ''
+  if (this.$textarea.value !== this.oldValue) {
+    this.oldValue = this.$textarea.value
+    var boundUpdateCountMessage = this.updateCountMessage.bind(this)
+    boundUpdateCountMessage()
+  }
+}
+
+// Update message box
+CharacterCount.prototype.updateCountMessage = function () {
+  var countElement = this.$textarea
+  var options = this.options
+  var countMessage = this.countMessage
+
+  // Determine the remaining number of characters/words
+  var currentLength = this.count(countElement.value, options)
+  var maxLength = this.maxLength
+  var remainingNumber = maxLength - currentLength
+
+  // Set threshold if presented in options
+  var threshold = 0
+  if (options.threshold) {
+    threshold = options.threshold
+  }
+  var thresholdValue = maxLength * threshold / 100
+  if (thresholdValue > currentLength) {
+    countMessage.classList.add('govuk-character-count__message--disabled')
+  } else {
+    countMessage.classList.remove('govuk-character-count__message--disabled')
+  }
+
+  // Update styles
+  if (remainingNumber < 0) {
+    countElement.classList.add('govuk-textarea--error')
+    countMessage.classList.remove('govuk-hint')
+    countMessage.classList.add('govuk-error-message')
+  } else {
+    countElement.classList.remove('govuk-textarea--error')
+    countMessage.classList.remove('govuk-error-message')
+    countMessage.classList.add('govuk-hint')
+  }
+
+  // Update message
+  var charVerb = 'remaining'
+  var charNoun = 'character'
+  var displayNumber = remainingNumber
+  if (options.maxwords) {
+    charNoun = 'word'
+  }
+  charNoun = charNoun + ((remainingNumber === -1 || remainingNumber === 1) ? '' : 's')
+
+  charVerb = (remainingNumber < 0) ? 'too many' : 'remaining'
+  displayNumber = Math.abs(remainingNumber)
+
+  countMessage.innerHTML = 'You have ' + displayNumber + ' ' + charNoun + ' ' + charVerb
+}
+
+CharacterCount.prototype.handleFocus = function () {
+  // Check if value changed on focus
+  this.valueChecker = setInterval(this.checkIfValueChanged.bind(this), 1000)
```

Comment:
I wonder if we can take advantage of this possible alternative:

https://github.com/alphagov/accessible-autocomplete/issues/164#issuecomment-392875126

Reply:
Good suggestion. I'll try.

Reply:
Hmm, does not support IE8. But we could be OK with this since it'll likely only be used for users of assistive technologies that we can assume will be using a newer version of IE?

Reply:
Tried it and behaves as expected (starts on focus, stops on blur). Could do a second test to be sure.

Reply:
I'll raise an issue after we merge this to investigate the approach defined in that issue.